src/components/list-item/list-item.tsx

```diff
@@ -147,44 +271,161 @@ export class ListItem implements ConditionalSlotComponent, InteractiveComponent
+          aria-posinset={posInSet}
+          aria-setsize={setSize}
+          class={CSS.container}
+          onFocus={this.focusCellNull}
+          onKeyDown={this.handleItemKeyDown}
+          ref={(el) => (this.containerEl = el)}
+          role="row"
+          tabIndex={active ? 0 : -1}
+        >
+          {this.renderExpand()}
           {this.renderActionsStart()}
           {this.renderContentContainer()}
           {this.renderActionsEnd()}
-        </div>
-        <div class={CSS.nestedContainer}>
-          <slot />
+        </tr>
+        <div
+          class={{
+            [CSS.nestedContainer]: true,
+            [CSS.nestedContainerHidden]: expandable ? !expanded : false
+          }}
+        >
+          <slot onSlotchange={this.handleDefaultSlotChange} />
         </div>
       </Host>
     );
   }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  handleDefaultSlotChange = (event: Event): void => {
+    const listItemChildren = getListItemChildren(event);
+    updateListItemChildren(listItemChildren);
+    this.expandable = !!listItemChildren.length;
+  };
+
+  toggleExpanded = (): void => {
+    this.expanded = !this.expanded;
+  };
+
+  handleItemContentKeyDown = (event: KeyboardEvent): void => {
+    const { key } = event;
+
+    if (key === " " || key === "Enter") {
+      event.preventDefault();
+      this.emitListItemSelect();
+    }
+  };
+
+  handleItemKeyDown = (event: KeyboardEvent): void => {
+    const { key } = event;
+    const composedPath = event.composedPath();
+    const { containerEl, contentEl, actionsStartEl, actionsEndEl, expanded, expandable } = this;
+
+    const cells = [actionsStartEl, contentEl, actionsEndEl].filter(Boolean);
+    const currentIndex = cells.findIndex((cell) => composedPath.includes(cell));
+
+    if (key === "ArrowRight") {
+      event.preventDefault();
+      const nextIndex = currentIndex + 1;
+      if (currentIndex === -1) {
+        if (!expanded && expandable) {
+          this.expanded = true;
+          this.focusCell(null);
+        } else if (cells[0]) {
+          this.focusCell(cells[0]);
+        }
+      } else if (cells[currentIndex] && cells[nextIndex]) {
+        this.focusCell(cells[nextIndex]);
+      }
+    } else if (key === "ArrowLeft") {
+      event.preventDefault();
+      const prevIndex = currentIndex - 1;
+      if (currentIndex === -1) {
+        this.focusCell(null);
+        if (expanded && expandable) {
+          this.expanded = false;
+        } else {
+          this.calciteInternalFocusPreviousItem.emit();
+        }
+      } else if (currentIndex === 0) {
+        this.focusCell(null);
+        containerEl.focus();
+      } else if (cells[currentIndex] && cells[prevIndex]) {
+        this.focusCell(cells[prevIndex]);
+      }
+    }
+  };
+
+  emitListItemSelect = (): void => {
```

Comment:
I'm still not sold on having a "selection" built into list. Ideally, a user could just slot a div or whatever as their content and add their own onclick to that. That way we aren't having to implement any selection.

The problem currently, is that we have `description` and `label` properties for defining list content.

It may be worth creating a component for those (heading/description) so that users can slot it in other places too.

Reply:
> I can see selection being a common use for the list, it'd be convenient, and help w/ consistency, for users to have the hook built in. Any other concerns aside from not having selection code?

If that is the case, should selection be built in where we control the UI? Currently, its just emitting an event on click. Thats what I mean by selection.